File: backend/routers/wb_sales.py
"""Сырые продажи — порт WbSalesController::actionIndex/View (образец: routers/cost.py)."""
from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.ext.asyncio import AsyncSession

from backend.database import get_db
from backend.deps import get_current_company
from backend.services.wb_sales_service import WbSalesService

router = APIRouter(prefix="/api/wb-sales", tags=["wb-sales"])


@router.get("")
async def wb_sales_list(
    date_from: str | None = Query(default=None),
    date_to: str | None = Query(default=None),
    nm_id: int | None = Query(default=None, ge=1),
    income_id: int | None = Query(default=None),
    is_supply: int | None = Query(default=None, ge=0, le=1),
    is_realization: int | None = Query(default=None, ge=0, le=1),
    total_price: float | None = Query(default=None),
    finished_price: float | None = Query(default=None),
    sale_id: str | None = Query(default=None),
    srid: str | None = Query(default=None),
    number: str | None = Query(default=None),
    supplier_article: str | None = Query(default=None),
    barcode: str | None = Query(default=None),
    warehouse_name: str | None = Query(default=None),
    warehouse_type: str | None = Query(default=None),
    country_name: str | None = Query(default=None),
    oblast_name: str | None = Query(default=None),
    region_name: str | None = Query(default=None),
    subject: str | None = Query(default=None),
    category: str | None = Query(default=None),
    brand: str | None = Query(default=None),
    card_title: str | None = Query(default=None),
    geo: str | None = Query(default=None),
    sort: str | None = Query(default=None),
    page: int = Query(default=1, ge=1),
    page_size: int = Query(default=20, ge=1, le=200),
    db: AsyncSession = Depends(get_db),
    company_id: int | None = Depends(get_current_company),
):
    svc = WbSalesService(db, company_id=company_id)
    return await svc.list(date_from, date_to, nm_id, income_id, is_supply,
                          is_realization, total_price, finished_price, sale_id,
                          srid, number, supplier_article, barcode,
                          warehouse_name, warehouse_type, country_name,
                          oblast_name, region_name, subject, category, brand,
                          card_title, geo, sort, page, page_size)


@router.get("/{sale_id}")
async def wb_sales_detail(
    sale_id: str,
    db: AsyncSession = Depends(get_db),
    company_id: int | None = Depends(get_current_company),
):
    svc = WbSalesService(db, company_id=company_id)
    row = await svc.get(sale_id)
    if not row:
        raise HTTPException(status_code=404, detail="Продажа не найдена")
    return row

File: frontend/src/api/wbSales.ts
import { api } from './client'

export interface WbSaleRow {
  saleID: string
  date: string | null
  nmId: number | null
  supplierArticle: string | null
  cardTitle: string | null
  totalPrice: number | null
  discountPercent: number | null
  priceWithDisc: number | null
  spp: number | null
  finishedPrice: number | null
  forPay: number | null
  warehouseName: string | null
  warehouseType: string | null
  countryName: string | null
  oblastOkrugName: string | null
  regionName: string | null
  [k: string]: unknown
}

export interface WbSaleListParams {
  date_from?: string
  date_to?: string
  nm_id?: number
  income_id?: number
  is_supply?: number | string
  is_realization?: number | string
  total_price?: number
  finished_price?: number
  sale_id?: string
  srid?: string
  number?: string
  supplier_article?: string
  barcode?: string
  warehouse_name?: string
  warehouse_type?: string
  country_name?: string
  oblast_name?: string
  region_name?: string
  subject?: string
  category?: string
  brand?: string
  card_title?: string
  geo?: string
  sort?: string
  page?: number
  page_size?: number
}

export const wbSalesApi = {
  async list(params: WbSaleListParams): Promise<{ items: WbSaleRow[]; total: number; page: number; page_size: number }> {
    const { data } = await api.get('/api/wb-sales', { params })
    return data
  },
  async get(saleID: string): Promise<WbSaleRow> {
    const { data } = await api.get(`/api/wb-sales/${encodeURIComponent(saleID)}`)
    return data
  },
}
